postgres-appliance/bootstrap/clone_with_basebackup.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+import argparse
+from collections import namedtuple
+import logging
+import os
+import subprocess
+
+logging.basicConfig(format='%(asctime)s %(levelname)s: %(message)s', level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+def read_configuration():
+    parser = argparse.ArgumentParser(description="Script to clone from another cluster using pg_basebackup")
+    parser.add_argument('--scope', required=True, help='target cluster name')
+    parser.add_argument('--datadir', required=True, help='target cluster postgres data directory')
+    parser.add_argument('--bindir', required=True, help='the directory with pg_basebackup')
+    parser.add_argument('--from-pgpass', required=True, help='path to the pgpass file containing credentials for the instance to be cloned')
+    args = parser.parse_args()
+
+    options = namedtuple('Options', 'name datadir bindir pgpassfile')
+
+    result=options(name=args.scope, datadir=args.datadir, bindir=args.bindir, pgpassfile=args.from_pgpass)
+    return result
+
+def parse_pgpass(pgpassfile):
+    with open(pgpassfile, 'r') as f:
+        line = f.readline()
+        fields = line.strip().split(':')
+        if len(fields) != 5:
+            raise Exception("pgpass file must be in a format 'hostname:port:database:username:password'")
+        return namedtuple('pgpass', 'host port dbname user')._make(fields[:4])
+
+def escape_value(val):
+    quote = False
+    temp = []
+    for c in val:
+        if c.isspace():
+            quote = True
+        elif c in ('\'', '\\'):
+            temp.append('\\')
+        temp.append(c)
+    result = ''.join(temp)
+    return result if not quote else '\'{0}\''.format(result)
+
+def prepare_connection(options):
+    pgpass = parse_pgpass(options.pgpassfile)
```

Comment:
I use pgpass to pass the password for the connection, and since pgpass requires exact match of other fields as well, and those are exact fields required to connect to the cluster, it makes sense to read all of them from pgpass.